test/system/src/test/java/io/pravega/test/system/TransactionStressTest.java

```diff
@@ -0,0 +1,390 @@
+        assertTrue(deleteScopeStatus.get());
+        log.info("Test {} succeeds ", "MultiReaderWriterTxnWithFailOver");
+    }
+
+
+    private CompletableFuture<Void> startReading(final ConcurrentLinkedQueue<Long> readResult, final AtomicLong writeCount, final
+    AtomicLong readCount, final AtomicBoolean exitFlag, final EventStreamReader<Long> reader) {
+        return CompletableFuture.runAsync(() -> {
+            log.info("Exit flag status {}", exitFlag);
+            log.info("Read count {} and write count {}", readCount.get(), writeCount);
+            while (!(exitFlag.get() && readCount.get() == writeCount.get())) {
+                log.info("Entering read loop");
+                // exit only if exitFlag is true  and read Count equals write count.
+                try {
+                    final Long longEvent = reader.readNextEvent(SECONDS.toMillis(60)).getEvent();
+                    log.debug("Reading event {}", longEvent);
+                    if (longEvent != null) {
+                        //update if event read is not null.
+                        readResult.add(longEvent);
+                        readCount.incrementAndGet();
+                        log.debug("Event read count {}", readCount);
+                    } else {
+                        log.debug("Read timeout");
+                    }
+                } catch (Throwable e) {
+                    log.error("Test Exception while reading from the stream: ", e);
+                }
+            }
+
+        }, executorService);
+    }
+
+    private CompletableFuture<Void> startWritingIntoTxn(final AtomicLong data, EventStreamWriter<Long> writer,
+                                                        final AtomicLong eventWriteCount) {
+        return CompletableFuture.runAsync(() -> {
+            while (this.transactionCount.incrementAndGet() < NUM_TRANSACTIONS) {
+                Transaction<Long> transaction = null;
+                try {
+                    transaction = retry
+                            .retryingOn(TxnCreationFailedException.class)
+                            .throwingOn(RuntimeException.class)
+                            .run(() -> createTransaction(writer));
+
+                    //each transaction has EVENTS_PER_TRANSACTION events
+                    for (int j = 0; j < EVENTS_PER_TRANSACTION; j++) {
+                        long value = data.incrementAndGet();
+                        transaction.writeEvent(String.valueOf(value), value);
+                        log.debug("Writing event: {} into transaction: {}", value, transaction);
+                    }
+                    //commit Txn
+                    transaction.commit();
+
+                    //wait for transaction to get committed
+                    checkTxnStatus(transaction, eventWriteCount).get();
+                } catch (Throwable e) {
+                    log.warn("Exception while writing events in the transaction: {}", e);
```

Comment:
I think I know why you are splitting the log message into two parts, but given that this is a test and we shouldn't be getting an exception often, let's just log the whole thing as part of the warn.

Also, I noticed that we aren't being consistent and in line 313 we have an error instead off a warn. I think they are both warns because we are not exiting the loop because of the exceptions. If they were errors, then I suppose we would be exiting the while loop.